Applications/WebPortal/WebApp/library/source/pages/account_directory_page/group_selection_box.tsx

```diff
@@ -0,0 +1,500 @@
+  showSuggestions: boolean;
+}
+
+/** Displays a box that shows suggested groups based on the input string. */
+export class GroupSelectionBox extends React.Component<Properties, State> {
+  public static readonly defaultProps = {
+    value: '',
+    selectedGroups: [] as Beam.DirectoryEntry[],
+    suggestions: [] as Beam.DirectoryEntry[],
+    isError: false,
+    onAddGroup: () => {},
+    onRemoveGroup: () => {},
+    onValueChange: () => {}
+  };
+
+  constructor(props: Properties) {
+    super(props);
+    this.state = {
+      currentIndex : 0,
+      showSuggestions: false
+    };
+    this.inputRef = React.createRef();
+    this.onFocus = this.onFocus.bind(this);
+    this.onBlur = this.onBlur.bind(this);
+    this.onAddGroup = this.onAddGroup.bind(this);
+    this.onChangeIndex = this.onChangeIndex.bind(this);
+    this.onKeyPress = this.onKeyPress.bind(this);
+  }
+
+  public render(): JSX.Element {
+    const inputStyle = (() => {
+      if(this.props.displaySize === DisplaySize.SMALL) {
+        return GroupSelectionBox.DYNAMIC_STYLE.inputSmall;
+      } else if(this.props.displaySize === DisplaySize.MEDIUM) {
+        return GroupSelectionBox.DYNAMIC_STYLE.inputMedium;
+      } else {
+        return GroupSelectionBox.DYNAMIC_STYLE.inputLarge;
+      }
+    })();
+    const errorBoxStyle = (() => {
+      if(this.props.isError) {
+        return GroupSelectionBox.DYNAMIC_STYLE.error;
+      } else {
+        return null;
+      }
+    })();
+    const textStyle = (() => {
+      if(this.props.displaySize === DisplaySize.SMALL) {
+        return GroupSelectionBox.DYNAMIC_STYLE.textSmall;
+      } else {
+        return GroupSelectionBox.DYNAMIC_STYLE.textLarge;
+      }
+    })();
+    const suggestionBox = (() => {
+      if(this.state.showSuggestions) {
+        return GroupSelectionBox.STYLE.suggestionWrapper;
+      } else {
+        return GroupSelectionBox.STYLE.hidden;
+      }
+    })();
+    const selectedGroups = this.props.selectedGroups.map((group) => {
+        return (<div className={css(GroupSelectionBox.DYNAMIC_STYLE.groupEntry)}
+            key={group.id}>
+          <div className={css(GroupSelectionBox.DYNAMIC_STYLE.textLarge)}>
+            {group.name}
+          </div>
+          <div style={GroupSelectionBox.STYLE.imageWrapper}>
+            <img style={GroupSelectionBox.STYLE.image}
+              onClick={ () => this.props.onRemoveGroup(group) }
+              src={'resources/remove.svg'}/>
+          </div>
+        </div>);
+    });
+  return (
+    <div id='GROUP BOX' style={GroupSelectionBox.STYLE.box}
+        onBlur={this.onBlur}
+        onFocus={this.onFocus}>
+      <input type='text'
+        ref={this.inputRef}
+        value={this.props.value}
+        onKeyDown={(event: React.KeyboardEvent<HTMLInputElement>) =>
+          this.onKeyPress(event)}
+        onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
+          this.props.onValueChange(event.target.value);}}
```

Comment:
Use a method for this to avoid constantly constructing lambda expressions on every render.